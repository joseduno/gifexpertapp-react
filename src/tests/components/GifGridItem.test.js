import { shallow } from "enzyme"
import GifGridItem from '../../components/GifGridItem'


describe('Pruebas en el componentes GifGridItem', () => {

    const title = 'Testing'
    const url = 'testing.com'
    const wrapper = shallow(<GifGridItem title={title} url={url} />)

    test('Debe mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('debe tener un parrafo con el titulo', () => {
        const p = wrapper.find('p')
        expect(p.text().trim()).toBe(title)
    })

    test('debe tener imagen igual a url y alt de los props', () => {
        const img = wrapper.find('img')
        expect(img.props().src).toBe(url)  // tambien funciona img.prop('src')
        expect(img.props().alt).toBe(title)
    })

    test('debe tener la clase animate__flipInX', () => {
        const div = wrapper.find('div')
        const validation = div.prop('className').includes('animate__flipInX')
        expect(validation).toBe(true)
    })
    
})
