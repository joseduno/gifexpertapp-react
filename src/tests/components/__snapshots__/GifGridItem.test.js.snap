// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el componentes GifGridItem Debe mostrar el componente correctamente 1`] = `
<div
  className="card animate__animated animate__flipInX"
>
  <img
    alt="Testing"
    src="testing.com"
  />
  <p>
    Testing
  </p>
</div>
`;
